searx/engines/livespace.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""LiveSpace (Videos)
+
+.. hint::
+
+   This engine only search for **live streams**!
+
+"""
+
+from urllib.parse import urlencode
+from datetime import datetime
+from babel import dates
+
+about = {
+    "website": 'https://live.space',
+    "wikidata_id": None,
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+categories = ['videos']
+
+base_url = 'https://backend.live.space'
+
+# engine dependent config
+paging = True
+results_per_page = 10
+
+
+def request(query, params):
+
+    args = {'page': params['pageno'] - 1, 'searchKey': query, 'size': results_per_page}
+    params['url'] = f"{base_url}/search/public/stream?{urlencode(args)}"
+    params['headers'] = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+
+    return params
+
+
+def response(resp):
+
+    results = []
+    json = resp.json()
+    now = datetime.now()
+
+    # for live videos
+
+    for result in json.get('result', []):
+
+        title = result.get("title")
+        thumbnailUrl = result.get("thumbnailUrl")
+        category = result.get("category/name")
+        username = result.get("user", {}).get("userName", "")
+        url = f'https://live.space/watch/{username}'
+
+        # stream tags
+        # currently the api seems to always return null before the first tag,
+        # so strip that unless it's not already there
+        tags = ''
+        if result.get("tags"):
+            tags = [x for x in result.get("tags").split(';') if x and x != 'null']
+            tags = ', '.join(tags)
+
+        content = []
+        if category:
+            content.append(f'category - {category}')
+
+        if tags and len(tags) > 0:
+            content.append(f'tags - {tags}')
+
+        # time & duration
+        start_time = None
+        if result.get("startTimeStamp"):
+            start_time = datetime.fromtimestamp(result.get("startTimeStamp") / 1000)
+
+        # for VODs (videos on demand)
+        end_time = None
+        if result.get("endTimeStamp"):
+            end_time = datetime.fromtimestamp(result.get("endTimeStamp") / 1000)
+
+        timestring = ""
+        if start_time:
+            delta = (now if end_time is None else end_time) - start_time
+            timestring = dates.format_timedelta(delta, granularity='second')
```

Comment:
@Bnyro aked ..

> I'm not sure if we already have a pre-build function for formatting time span strings. (maybe @return42 knows? :) )

Here I used [babel.dates.format_timedelta](https://babel.pocoo.org/en/latest/api/dates.html#babel.dates.format_timedelta).

Reply:
That's nice, thanks :+1:

I thought I've seen a manual implementation of this for an engine once, I'll see if I can find it (if it exists) and replace it too.